src/main/java/emissary/util/TimeUtil.java

```diff
@@ -232,6 +234,53 @@ public static ZonedDateTime getZonedDateFromISO8601(@Nullable final String dateS
         return dateString == null ? null : ZonedDateTime.parse(dateString, DATE_ISO_8601);
     }
 
+    /**
+     * Parses hex date string into formatted string ("yyyy-MM-dd HH:mm:ss.SSS")
+     *
+     * @param hexDate hex string representation of the date
+     * @return converted time in string format yyyy-MM-dd HH:mm:ss.SSS
+     * @throws DateTimeParseException if could not parse date
+     *
+     */
+    public static String convertHexDate(String hexDate) {
+        String hexNumber = hexDate.substring(2);
+        String dateHex = hexNumber.substring(0, 8);
+        String timeHex = hexNumber.substring(8, 16);
+
+        long daysToAdd = Long.parseLong(dateHex, 16);
+        long millisToAdd = Math.round(Long.parseLong(timeHex, 16) * 10 / 3.0);
```

Comment:
Just curious - anybody understand why this is dividing by 3.0?   The source from which we're migrating this is woefully uncommented.